url863d2ab553_themadcreator_delaunay_tgz-pJ8-org_delaunay_PoissonDiscSamplesDemoJ8/src/org/delaunay/dtfe/IsolineBuilder.java

```diff
@@ -0,0 +1,139 @@
+package org.delaunay.dtfe;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+import org.delaunay.dtfe.DtfeTriangulationMap.ScaleType;
+import org.delaunay.model.Edge;
+import org.delaunay.model.Triangle;
+import org.delaunay.model.Vector;
+import org.delaunay.model.Vertex;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Sets;
+
+public class IsolineBuilder {
+
+	private final DtfeTriangulationMap<? extends DensityModel> dtfe;
+	private ScaleType scaleType = ScaleType.LOG;
+	
+	public IsolineBuilder(DtfeTriangulationMap<? extends DensityModel> dtfe) {
+		this.dtfe = dtfe;
+	}
+	
+	public IsolineBuilder setScaleType(ScaleType scaleType) {
+		this.scaleType = scaleType;
+		return this;
+	}
+
+	/**
+	 * Returns a list of list of vectors. Each list of vectors is a continuous
+	 * isoline at the specified value.
+	 */
+	public List<List<Vector>> getIsoLines(double value) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 42-72 into a separate method.